config/property/hazelcast_properties.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package property
+
+import (
+	"os"
+
+	"strings"
+
+	"time"
+
+	"fmt"
+
+	"github.com/hazelcast/hazelcast-go-client/config"
+)
+
+// HazelcastProperties is a container for configured Hazelcast properties.
+// A Hazelcast property can be set via:
+//  * an environmental variable
+//  * programmatic configuration Config.SetProperty
+type HazelcastProperties struct {
+	properties config.Properties
+}
+
+// NewHazelcastPropertiesWithConfig returns HazelcastProperties with the given config's properties.
+func NewHazelcastPropertiesWithConfig(config *config.Config) *HazelcastProperties {
```

Comment:
This constructor does not seem really necessary. I think caller should call `NewHazelcastPropertiesWithProperties` , and pass the properties instead of config always.